src/main/java/com/yelp/nrtsearch/server/luceneserver/field/BooleanFieldDef.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2020 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver.field;
+
+import static com.yelp.nrtsearch.server.luceneserver.analysis.AnalyzerCreator.hasAnalyzer;
+
+import com.yelp.nrtsearch.server.grpc.FacetType;
+import com.yelp.nrtsearch.server.grpc.Field;
+import com.yelp.nrtsearch.server.luceneserver.doc.LoadedDocValues;
+import java.io.IOException;
+import java.util.List;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.FieldType;
+import org.apache.lucene.document.NumericDocValuesField;
+import org.apache.lucene.document.SortedNumericDocValuesField;
+import org.apache.lucene.index.DocValues;
+import org.apache.lucene.index.DocValuesType;
+import org.apache.lucene.index.IndexOptions;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.NumericDocValues;
+import org.apache.lucene.index.SortedNumericDocValues;
+
+/** Field class for 'BOOLEAN' field type. */
+public class BooleanFieldDef extends IndexableFieldDef {
+  protected BooleanFieldDef(String name, Field requestField) {
+    super(name, requestField);
+  }
+
+  @Override
+  protected void validateRequest(Field requestField) {
+    super.validateRequest(requestField);
+
+    if (requestField.getHighlight()) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Field: %s cannot have highlight=true. only type=text or type=atom fields can have highlight=true",
+              getName()));
+    }
+
+    if (hasAnalyzer(requestField) && !requestField.getSearch()) {
+      throw new IllegalArgumentException("No analyzer allowed when search=false");
+    }
+  }
+
+  @Override
+  protected DocValuesType parseDocValuesType(Field requestField) {
+    if (requestField.getStoreDocValues() || requestField.getSort() || requestField.getGroup()) {
+      if (requestField.getMultiValued()) {
```

Comment:
Up for discussion: Do we want to support multiple values for boolean fields, or validate and enforce single values?

Reply:
I would lean towards disallowing multivalued boolean fields altogether. However, I would punt this until later. The boolean type generally needs some work. It currently can't even be used in term queries.

Reply:
Sounds good, we could do this later.